### PR TITLE
Add DeepSeek ckpt conversion script

### PR DESCRIPTION
**Pull request by RissyRan**
# Description

Add DeepSeek ckpt conversion script for DeepSeek v2-16b and v3-671b.

# Tests

* Tested it locally with v2-16b in training to compare logits

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: MaxText/convert_deepseek_ckpt.py [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+        "qk_nope_head_dim": 128,
+        "qk_rope_head_dim": 64,
+        "v_head_dim": 128,
+    },
+}
+
+
+# skip to convert ckpts related to fp8 and MTP for now
+SKIPPED_KEYS_SUFFIX = [
+    "_scale_inv",
```
Comment (gagika) [inline MaxText/convert_deepseek_ckpt.py:72]: I don't think we need this `_scale_inv` here, as this script don't work on fp8 checkpoint and for bf16 converted checkpoint `_scale_inv` .

It will be better if we removed `_scale_inv` check, so it will fail if we run on fp8 checkpoint (even better, if you could check "_scale_inv" suffix and provide an error message that the script shouldn't be run on fp8 checkpoint.

If you remove "_scale_inv" from the list, the rest refers to MTP, so perhaps you can update the name SKIPPED_KEYS_SUFFIX to MTP_KEYS_SUFFIX

**Review by gagika — APPROVED** [review]
LGTM, only one comment

File: MaxText/convert_deepseek_ckpt.py (hunk 1 above)
Comment (RissyRan) [reply MaxText/convert_deepseek_ckpt.py:72]: Sounds good! I will make the update.

Comment (RissyRan) [reply MaxText/convert_deepseek_ckpt.py]: Updated.

**Review by khatwanimohit — APPROVED** [review]
